kratos/geometries/line_2d_2.h

```diff
@@ -890,6 +890,57 @@ class Line2D2 : public Geometry<TPointType>
         return false;
     }
 
+    /** Test the intersection with another geometry
+     *  Test if this geometry intersects with other line_2d_2
+     *
+     * @param  rOtherGeometry Geometry to intersect with
+     * @return True if the geometries intersect, False in any other case.
+     */
+    virtual bool HasIntersection(const BaseType& rOtherGeometry) override
+    {
+        // We get the local points
+        const TPointType& first_point  = BaseType::GetPoint(0); //p1
+        const TPointType& second_point = BaseType::GetPoint(1); //p2
+
+        // We get the other line's points
+        const TPointType& first_point_other  = *rOtherGeometry(0); //p3
+        const TPointType& second_point_other = *rOtherGeometry(1); //p4
+
+        // parametric coordinate of intersection on current line
+        double t = ( (first_point[0]-first_point_other[0])*(first_point_other[1] - second_point_other[1]) - (first_point[1]-first_point_other[1])*(first_point_other[0]-second_point_other[0]) )
+                    /
+                   ( (first_point[0]-second_point[0])*(first_point_other[1] - second_point_other[1]) - (first_point[1]-second_point[1])*(first_point_other[0]-second_point_other[0]) );
+
+        return (0.0<=t) && (t<=1.0);
```

Comment:
no tolerance? Is this intended / how is it done for the other geometries?

Reply:
```suggestion
        return (t > -1.0e-12) && (t<(1.0 + 1.0e-12));
```

My suggestion